src/systemctl/systemctl.c

```diff
@@ -340,6 +340,12 @@ static bool output_show_unit(const UnitInfo *u, char **patterns) {
         if (arg_all)
                 return true;
 
+        /* Note that '--all' is not purely a state filter, but also a
+         * filter that hides units that "follow" other units (which is
+         * used for device units that appear under different names). */
+        if (u->following[0])
```

Comment:
(yeah, i know, the old code didn't do this right either, but let's fix this and do it the same way as everywhere else.